engine/chain/models/arc_error.go

```diff
@@ -0,0 +1,24 @@
+package chainmodels
+
+import "fmt"
+
+// ArcError represents an error returned by the ARC API when status code is 4xx.
```

Comment:
Do we also want to consider handling other errors than 4xx? If not could you tell me why we only handle 4xx structure here?

Reply:
That's the only one model from ARC documentation. 
They always returns it when "something-went-wrong".

So what other models you would suggest to handle?

Reply:
Ah okay, I thought they also return some "different" errors for example when a 5xx is thrown or something. If it's the only model than it's all good